doc/api/worker_threads.md

```diff
@@ -1527,6 +1527,79 @@ Calling `unref()` on a worker allows the thread to exit if this is the only
 active handle in the event system. If the worker is already `unref()`ed calling
 `unref()` again has no effect.
 
+## `worker.makeSync(buffer[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `buffer` {SharedArrayBuffer} A shared memory buffer to use for communication.
+* `options` {Object}
+  * `timeout` {number} The timeout in milliseconds for synchronous calls. **Default:** `5000`.
+  * `expandable` {boolean} Whether the buffer can be resized. **Default:** `true` if the buffer
+    supports `growable` option.
+* Returns: {Object} An object with synchronous methods mirroring those exposed through [`worker.wire()`][].
+
+Creates a synchronous API facade that communicates with a worker thread over a shared memory buffer.
+The worker thread must call [`worker.wire()`][] on the same buffer to register the methods that can be called.
+
+This function enables making synchronous calls to a worker thread, which is particularly useful
+when code requires blocking operations but still wants to benefit from the worker thread's isolation.
+
+```js
+const { Worker, makeSync } = require('node:worker_threads');
+
+// Create a SharedArrayBuffer for communication
+const buffer = new SharedArrayBuffer(1024, {
+  maxByteLength: 64 * 1024 * 1024,
+});
+
+// Create a worker, passing the buffer
+const worker = new Worker('worker-script.js', {
+  workerData: { buffer },
+});
+
+// Create a synchronous API facade
+const api = makeSync(buffer);
+
+// Call a method synchronously - this will block until the worker responds
+const result = api.methodName(arg1, arg2);
+```
```

Comment:
The API here feels a bit awkward. I wonder if some kind of wrapper to hide the `SharedArrayBuffer` would be helpful...

```js
// in the main thread
const { Worker, APIProxy } = require('node:worker_threads');

const api = new APIProxy({ min: 1024, max: 64 * 1024 * 1024 });
const worker = new Worker('...', { workerData: api });

// in the worker
const { workerData } = require('node:worker_threads');
workerData.bind({ foo() { });
```

Reply:
Yes, it 100% could be. I think we could actually do better:

```js
// in the main thread
const { Worker, SyncAPIProxy } = require('node:worker_threads');

const api = new APIProxy('myworker.js', { min: 1024, max: 64 * 1024 * 1024 });

api.echo()

// in the worker
export async function echo (...args) { return args }
```

Reply:
It would be nice if we can make the `{min, max}` part optional and use some sensible defaults which can be overridden by the user if they wish to.